src/Camera.cpp

```diff
@@ -58,3 +58,36 @@ void Camera::RotateAround(float deltaX, float deltaY)
 }
 
 
+void Camera::Zoom(bool in)
+{
+	const float zoomSpeed = 0.2;
+
+	glm::vec3 direction = zoomSpeed*(normalize(forward-pos));
+	if(!in){
+		direction = -direction;
+	}
+	pos = pos + direction;
+
+	UpdateViewMatrix();
+}
+
+void Camera::Panning(float deltaX, float deltaY)
+{
+	const float panningSpeed = 0.1;
+	glm::vec3 bi = normalize(cross(forward-pos, up));
```

Comment:
These need the namespace glm before the function calls. e.g., glm::normalize(glm::cross(...));

I'm surprised this compiles for you.

Reply:
d